labview/msg_headers.c

```diff
@@ -0,0 +1,270 @@
+  }
+
+  // Allocate memory for temp buffer
+  char* headers = (char*)malloc(required_buffer_size * sizeof(char));
+  float f32_value;
+  double f64_value;
+
+  int index = 0;
+  for (int i = 0; i < table->num_entries; i++) {
+    headers[index] = table->entries[i].value.kind;  // Character that indicates a type
+    index++;
+
+    memcpy(headers + index, table->entries[i].key.bytes, table->entries[i].key.len);
+    index += table->entries[i].key.len;
+    headers[index] = '=';  // Separator
+    index++;
+
+    switch (table->entries[i].value.kind) {
+      case AMQP_FIELD_KIND_I8:
+        headers[index] = table->entries[i].value.value.i8;
+        index++;
+        break;
+      case AMQP_FIELD_KIND_I64:
+        for (int j = 0; j < 8; j++) {
+          headers[index + j] = (table->entries[i].value.value.i64 >> (j * 8)) & 0xFF;
+        }
+        index += 8;
+        break;
+      case AMQP_FIELD_KIND_UTF8:
+      case AMQP_FIELD_KIND_BYTES:
+        memcpy(headers + index, table->entries[i].value.value.bytes.bytes, table->entries[i].value.value.bytes.len);
+        index += table->entries[i].value.value.bytes.len;
+        break;
+      case AMQP_FIELD_KIND_U8:
+        headers[index] = table->entries[i].value.value.u8;
+        index++;
+        break;
+      case AMQP_FIELD_KIND_I16:
+        headers[index] = table->entries[i].value.value.i16 & 0xFF;
+        headers[index + 1] = (table->entries[i].value.value.i16 >> 8) & 0xFF;
+        index += 2;
+        break;
+      case AMQP_FIELD_KIND_U16:
+        headers[index] = table->entries[i].value.value.i16 & 0xFF;
+        headers[index + 1] = (table->entries[i].value.value.i16 >> 8) & 0xFF;
+        index += 2;
+        break;
+      case AMQP_FIELD_KIND_I32:
+        for (int j = 0; j < 4; j++) {
+          headers[index + j] = (table->entries[i].value.value.i32 >> (j * 8)) & 0xFF;
+        }
+        index += 4;
+        break;
+      case AMQP_FIELD_KIND_U32:
+        for (int j = 0; j < 4; j++) {
+          headers[index + j] = (table->entries[i].value.value.u32 >> (j * 8)) & 0xFF;
+        }
+        index += 4;
+        break;
+      case AMQP_FIELD_KIND_U64:
+      case AMQP_FIELD_KIND_TIMESTAMP:
+        for (int j = 0; j < 8; j++) {
+          headers[index + j] = (table->entries[i].value.value.u64 >> (j * 8)) & 0xFF;
+        }
+        index += 8;
+        break;
+      case AMQP_FIELD_KIND_F32:
+        f32_value = table->entries[i].value.value.f32;
+        memcpy(headers + index, &f32_value, 4);
+        index += 4;
+        break;
+      case AMQP_FIELD_KIND_F64:
+        f64_value = table->entries[i].value.value.f64;
+        memcpy(headers + index, &f64_value, 8);
+        index += 8;
+        break;
+      default:
+        break;  // Unsupported kind, ignore
+    }
+
+    headers[index] = ';';  // Separator
+    index++;
+  }
+
+  // Replace the last ';' with Null-terminate the C string
+  //headers[index - 1] = '\0';
```

Comment:
make sense, why is it commented?

Reply:
because LabVIEW string is not a c-string and doesn't needs to be closed by '/0'. Even more - it causes the problem while decodeing the string on LabVIEW side. However, this code should be removed.

Reply:
ok, please add this in comment